src/source/Common/AwsV4Signer.c

```diff
@@ -117,6 +117,11 @@ STATUS signAwsRequestInfo(PRequestInfo pRequestInfo)
 
     CHK(pRequestInfo != NULL && pRequestInfo->pAwsCredentials != NULL, STATUS_NULL_ARG);
 
+    // signAwsRequestInfo is a public api function, if service is not specified default to "kinesisvideo" so no breaking changes are introduced to the api
+    if (pRequestInfo->service[0] == L'\0') {

```

Comment:
A small nit. I would not modify a structure that's passed to this layer by a higher-layer. Instead, I would define a pointer to char like

PCHAR pServiceName = (PCHAR) KINESIS_VIDEO_SERVICE_NAME; 

in the beginning of the function. Then, here, I would do something like

if (pRequestInfo->service[0] != L'\0') {
    pServiceName = pRequestInfo->service;
}

This means that any place where pRequestInfo->service is referenced needs to reference pServiceName.

This is a minor nit though

Reply:
Hmm, instead of modifying anything I could always do it just in time, ie just replace `pRequestInfo->service` everywhere with:
```c
pRequestInfo->service[0] == L'\0' ? KINESIS_VIDEO_SERVICE_NAME : pRequestInfo->service
```

It seems the way you are suggesting here would be tricky bc this is used by other functions `generateAwsSigV4Signature`, `generateCanonicalRequestString`, `generateCredentialScope`, `generateEncodedCredentials` that pass around `pRequestInfo`

If its a minor nit I'd prefer to leave as-is (if this is acceptable), unless there is a very simple solution

Reply:
I agree, it really is a minor issue and not worth adding more complexity.